handle-singlepart-upload.go

```diff
@@ -63,7 +63,8 @@ func ConvertSinglePartUploadtoMultiPartUpload(f *proxy.Flow) error {
 	// Encrypt data in body
 	encryptBody, err := encryptBytes(f.Request.Raw().Context(),
 		config.KmsResourceName,
-		f.Request.Body)
+		f.Request.Body,
+		string(f.Request.Header.Get("x-request-id")))
```

Comment:
Do not change the signature of encryptBytes and decryptByte in this way . Please capture timings outside of the method itself. or create a wrapper function (encryptBytesWithTiming).

Reply:
Pawan and I talked about this before. I suggested to leverage the context param. I think a wrapper function (like decorator in python. not sure if go has the equivalent) would be better as it'd work on the other functions.